string table
string[] row_names